front/pages/poke/[wId]/spaces/[spaceId]/apps/[appId]/index.tsx

```
@@ -37,17 +57,55 @@ export const getServerSideProps = withSuperUserAuthRequirements<{
       notFound: true,
     };
   }
+  const coreAPI = new CoreAPI(config.getCoreAPIConfig(), logger);
+
+  const specificationHashes = await coreAPI.getSpecificationHashes({
+    projectId: app.dustAPIProjectId,
+  });
+
+  let specification = JSON.parse(app.savedSpecification ?? "{}");
```

We have a `safeParseJSON`.

---

We need to throw an error if the spec is not parseable (should not happen), so rather keep JSON.parse